aikau/src/main/resources/alfresco/renderers/actions/UploadNewVersion.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (C) 2005-2015 Alfresco Software Limited.
+ *
+ * This file is part of Alfresco
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * <p>This module provides the configuration that can be used for rendering a [menu item]{@link module:alfresco/menus/AlfMenuItem}
+ * for the "Upload New Version" action. The main thing this provides is the "renderFilters" configuration that can be
+ * used to determine whether or not the action should be displayed based on the attributes of the "currentItem" (e.g. the Node)
+ * that the action is to be used against.</p>
+ * <p>It should be possible to upload a new version of a Node when the following is true:
+ * <ul><li>The Node is NOT locked (unless the current user owns the lock)</li>
+ * <li>The Node is NOT a working copy (unless the current user is the working copy owner)</li>
+ * <li>The Node is NOT locked where the type of lock is "NODE_LOCK"</li>
+ * </ul>
+ * 
+ * @module alfresco/renderers/actions/UploadNewVersion
```

Comment:
This isn't a class. It oughtn't really have a capital letter at the start (it's confusing and not best practice). Is this just "the way it is"?